src/Oscoin/Data/OscoinTx.hs

```diff
@@ -0,0 +1,333 @@
+{-# LANGUAGE DataKinds            #-}
+{-# LANGUAGE UndecidableInstances #-}
+{-# OPTIONS_GHC -fno-warn-deprecations #-}
+
+-- | The Transaction.
+--
+module Oscoin.Data.OscoinTx where
+
+import           Oscoin.Prelude
+
+import           Oscoin.Configuration (Network(..))
+import           Oscoin.Crypto.Address
+import           Oscoin.Crypto.Blockchain.Block (BlockHash)
+import qualified Oscoin.Crypto.Hash as Crypto
+import           Oscoin.Crypto.PubKey
+import           Oscoin.Data.Core
+
+import qualified Codec.Serialise as CBOR
+import qualified Crypto.Data.Auth.Tree as WorldState
+import           Data.ByteArray (ByteArrayAccess)
+import qualified Data.Map as Map
+
+-------------------------------------------------------------------------------
+-- Tx
+-------------------------------------------------------------------------------
+
+-- | A transaction on the ledger
+data Tx' (version :: Nat) (flags :: Nat) c = Tx'
+    { txVersion :: Word16
+    -- ^ Transaction version. Must be @1@.
+    , txFlags   :: Word16
+    -- ^ Feature flags. Must be @0@.
+    , txPayload :: TxPayload c
```

Comment:
`txMessage`?

Reply:
It used to be that, but I've seen transactions as a whole being called "messages", so not sure :thinking:

Reply:
I’m also fine with `txPayload`.